runtime/util/vmargs.c

```diff
@@ -76,8 +76,41 @@
 
 #define LARGE_STRING_BUF_SIZE 256
 
+struct JavaLauncherOption {
+        char *optionName;
+};
+
+/* options in this table will be not treated as a user Argument */
+static const struct JavaLauncherOption JavaLauncherOptionTable[] = {
+	"-Djava.class.path=",
+	"-Dsun.java.command=",
+	"-Dsun.java.launcher.pid=",
+	"-Dsun.java.launcher=",
+	"-Xjcl:jclse11_29",
+};
+
+#define ignoredOptionTableEnd (sizeof(JavaLauncherOptionTable) / sizeof(struct JavaLauncherOption) - 1)
+
 static char * getStartOfOptionValue(J9VMInitArgs* j9vm_args, IDATA element, const char *optionName);
 
+UDATA
+findArgInIgnoreTable(const char *optionName, IDATA start, IDATA end) {
+	IDATA mid = (start + end) / 2;
+	if (strncmp(JavaLauncherOptionTable[mid].optionName, optionName, strlen(JavaLauncherOptionTable[mid].optionName)) == 0) {
+	/* The search function found the given option name in the table. */
+		return TRUE;
+	} else if (start == end) {
+		/* The search function did not find the given option name in the table. */
+		return FALSE;
+	} else if (strncmp(JavaLauncherOptionTable[mid].optionName, optionName, strlen(JavaLauncherOptionTable[mid].optionName)) < 0) {
```

Comment:
Please precompute the lengths and put them in a separate table.  `sizeof(theString) -1` is the safest way.  Use `#define`s so you don't need to type the string twice.

Reply:
Should I put the precomputed length in a separate table? it might be easier to include them in the `JavaLauncherOption` object...

Reply:
> it might be easier to include them in the JavaLauncherOption object

Good idea.